### PR TITLE
Reduce market selector

### PR DESCRIPTION
**Pull request by bthaile**
figure this out.
https://github.com/AugurProject/augur/issues/1918

trying to help this issue
https://github.com/AugurProject/augur/issues/1823

pulling apart market selector it's too massive and heavy.

### CODE REVIEW

File: src/modules/portfolio/selectors/select-markets-open-orders.js [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
 
-export const marketsOpenOrders = createSelector(getOpenOrders, openOrders => {
-  const markets = openOrders.filter(
-    market => market.marketStatus !== constants.MARKET_CLOSED
-  );
+export const marketsOpenOrders = createSelector(selectMarkets, allMarkets => {
+  const markets = allMarkets.reduce((p, m) => {
+    if (m.marketStatus === constants.MARKET_CLOSED) return p;
+    const userOpenOrders = getUserOpenOrders(m.id);
+    if (userOpenOrders.length === 0) return p;
```
Comment (phoebemirman) [inline src/modules/portfolio/selectors/select-markets-open-orders.js]: Screen Shot 2019-04-26 at 2 32 30 PM
got this error here

**Comment by bthaile** [comment]
updated to protect against user open orders being undefined

**Comment by AugurIntegration** [comment]
This pull request **fixes 1 alert** when merging a706777c7c965f79862531fc24b5501e5aea571c into 3c3b04f61e495efda0b75e1aa1a38a4ad9cb5c41 - [view on LGTM.com](https://lgtm.com/projects/g/AugurProject/augur-ui/rev/pr-6f2901094a0af27c7c0862efea853203f71b5ea2)

**fixed alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/modules/orders/selectors/filled-orders.js [hunk 2]
```diff
@@ -54,7 +64,7 @@ function findOrders(
       const priceBN = createBigNumber(price);
       let typeOp = type;
 
-      const outcomeName = outcomesData[outcome].name;
+      const outcomeName = getOutcomeName(marketsData, outcomesData[outcome]);
```
Comment (phoebemirman) [inline src/modules/orders/selectors/filled-orders.js]: Screen Shot 2019-04-26 at 4 17 02 PM
got this here

Comment (bthaile) [reply src/modules/orders/selectors/filled-orders.js]: I think that occurs when outcomesData hasn't been loaded. I'll put in a guard

**Comment by AugurIntegration** [comment]
This pull request **fixes 1 alert** when merging 4693973ccd0df093b22d4ffd1258875355199202 into 2f9808c268f96d61e5a6070ee5dda55ef2a6dab2 - [view on LGTM.com](https://lgtm.com/projects/g/AugurProject/augur-ui/rev/pr-c13b2413afea04085d54cc102e87df4bea9da49b)

**fixed alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/modules/portfolio/helpers/get-last-trade-timestamp.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { convertUnixToFormattedDate } from "utils/format-date";
+import { ZERO } from "modules/common-elements/constants";
+import { createBigNumber } from "utils/create-big-number";
+
+export const getLastTradeTimestamp = marketTradHistory =>
```
Comment (phoebemirman) [inline src/modules/portfolio/helpers/get-last-trade-timestamp.js]: Typo here. Missing the E

**Comment by phoebemirman** [comment]
i also get this error when trying to sort by most recently traded in my created markets
Screen Shot 2019-04-26 at 5 50 15 PM

**Comment by bthaile** [comment]
Ok, will check out. I prob missed a place pitting recentlyTraded on the market object thats passed in.

**Comment by phoebemirman** [comment]
also get this error when portfolio info hasn't loaded

Screen Shot 2019-04-26 at 5 52 09 PM

**Comment by bthaile** [comment]
There are prob a bunch of null states we’ve not handled because tooltip was blocking giving data time to load.

File: src/modules/portfolio/helpers/get-last-trade-timestamp.js (hunk 3 above)
Comment (bthaile) [reply src/modules/portfolio/helpers/get-last-trade-timestamp.js]: fixed

**Comment by AugurIntegration** [comment]
This pull request **fixes 1 alert** when merging 3baa29bedfda5c1b66e251b700114f8019d4a0c7 into 2f9808c268f96d61e5a6070ee5dda55ef2a6dab2 - [view on LGTM.com](https://lgtm.com/projects/g/AugurProject/augur-ui/rev/pr-75277d14863083e25bfe803277290ac31891d034)

**fixed alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by AugurIntegration** [comment]
This pull request **fixes 1 alert** when merging ec316eda2898bf3e35083fbe27fe090f52e53355 into 2f9808c268f96d61e5a6070ee5dda55ef2a6dab2 - [view on LGTM.com](https://lgtm.com/projects/g/AugurProject/augur-ui/rev/pr-fcceb94b7c4b73ed8aaa6c0d281469741e160e2b)

**fixed alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**